Build fails

<!--
BUGS: Please use this template.

SUPPORT REQUESTS/QUESTIONS: If you have a support request or question please
submit them to StackOverflow using the tags aurelia and aurelia-framework
http://stackoverflow.com/questions/tagged/aurelia
or the Aurelia Gitter https://gitter.im/aurelia/discuss

Blue Spire offers paid support agreements. Further information regarding paid support
may be obtained by emailing support@bluespire.com

Future support requests submitted here will be closed.

The HTML comments below are for your reference, and are not displayed
when your issue is submitted, feel free to leave them.

Choose one of the two headings, delete the other.
-->
**I'm submitting a bug report**

* **Library Version:**
aurelia-ux 0.1.14


**Please tell us about your environment:**
* **Operating System:**
OSX 10.11.5

* **Node Version:**
7.2.1
<!--
Minimum supported Node version is latest Node 4.x LTS
run `node -v`
-->

* **NPM Version:**
3.10.10
<!--
Minimum supported NPM version is 3.x
run `npm -v`
-->

* **JSPM OR Webpack AND Version**
JSPM 0.16.48
<!--
If using JSPM
run `jspm -v`
If using Webpack
run `webpack --help | grep webpack`
-->

* **Browser:**
N/A

* **Language:**
TypeScript 2.1.4 


**Current behavior:**
When running `npm run build` the following error is thrown, `npm install` and `jspm install` were run before attempting to run the build.

`$ npm run build`

```
> aurelia-ux@0.1.14 prebuild /Users/dan/Documents/spikes/aurelia-ux/ux-master
> cross-env rimraf dist

```

```
> aurelia-ux@0.1.14 build /Users/dan/Documents/spikes/aurelia-ux/ux-master
> concurrently "npm run build:amd" "npm run build:commonjs" "npm run build:es2015" "npm run build:system"
```

```
[0] 
[0] > aurelia-ux@0.1.14 prebuild:amd /Users/dan/Documents/spikes/aurelia-ux/ux-master
[0] > cross-env rimraf dist/amd
[0] 
```
```
[1] 
[1] > aurelia-ux@0.1.14 prebuild:commonjs /Users/dan/Documents/spikes/aurelia-ux/ux-master
[1] > cross-env rimraf dist/commonjs
[1] 
```
```
[2] 
[2] > aurelia-ux@0.1.14 prebuild:es2015 /Users/dan/Documents/spikes/aurelia-ux/ux-master
[2] > cross-env rimraf dist/es2015
[2] 
```
```
[3] 
[3] > aurelia-ux@0.1.14 prebuild:system /Users/dan/Documents/spikes/aurelia-ux/ux-master
[3] > cross-env rimraf dist/system
[3] 
```
```
[0] 
[0] > aurelia-ux@0.1.14 build:amd /Users/dan/Documents/spikes/aurelia-ux/ux-master
[0] > cross-env tsc --project tsconfig.build.json --outDir dist/amd --module amd
[0] 
```
```
[1] 
[1] > aurelia-ux@0.1.14 build:commonjs /Users/dan/Documents/spikes/aurelia-ux/ux-master
[1] > cross-env tsc --project tsconfig.build.json --outDir dist/commonjs --module commonjs
[1] 
```
```
[2] 
[2] > aurelia-ux@0.1.14 build:es2015 /Users/dan/Documents/spikes/aurelia-ux/ux-master
[2] > cross-env tsc --project tsconfig.build.json --outDir dist/es2015 --module es2015
[2] 
```
```
[3] 
[3] > aurelia-ux@0.1.14 build:system /Users/dan/Documents/spikes/aurelia-ux/ux-master
[3] > cross-env tsc --project tsconfig.build.json --outDir dist/system --module system
[3] 
```
```
[0] src/effects/paper-wave.ts(342,5): error TS2531: Object is possibly 'null'.
[0] src/styles/style-controller.ts(60,5): error TS2322: Type 'Node | null' is not assignable to type 'Node'.
[0]   Type 'null' is not assignable to type 'Node'.
[0] src/styles/style-resource.ts(34,12): error TS2322: Type 'Promise<StyleFactory>' is not assignable to type 'Promise<StyleResource>'.
[0]   Type 'StyleFactory' is not assignable to type 'StyleResource'.
[0]     Property 'styleObjectType' is private in type 'StyleFactory' but not in type 'StyleResource'.
```
```
[3] src/effects/paper-wave.ts(342,5): error TS2531: Object is possibly 'null'.
[3] src/styles/style-controller.ts(60,5): error TS2322: Type 'Node | null' is not assignable to type 'Node'.
[3]   Type 'null' is not assignable to type 'Node'.
[3] src/styles/style-resource.ts(34,12): error TS2322: Type 'Promise<StyleFactory>' is not assignable to type 'Promise<StyleResource>'.
[3]   Type 'StyleFactory' is not assignable to type 'StyleResource'.
[3]     Property 'styleObjectType' is private in type 'StyleFactory' but not in type 'StyleResource'.
```
```
[0] 
[3] 
[0] npm
[0]  ERR! Darwin 15.5.0
[0] npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build:amd"
[0] npm ERR! node v7.2.1
[0] npm ERR! npm  v3.10.10
[0] npm ERR! 
[0] code ELIFECYCLE
[0] npm ERR! aurelia-ux@0.1.14 build:amd: `cross-env tsc --project tsconfig.build.json --outDir dist/amd --module amd`
[0] npm ERR! Exit status 2
[0] npm ERR! 
[0] npm ERR! Failed at the aurelia-ux@0.1.14 build:amd script 'cross-env tsc --project tsconfig.build.json --outDir dist/amd --module amd'.
[0] npm ERR! Make sure you have the latest version of node.js and npm installed.
[0] npm ERR! If you do, this is most likely a problem with the aurelia-ux package,
[0] npm ERR! not with npm itself.
[0] npm ERR!
[0]  Tell the author that this fails on your system:
[0] npm ERR!     cross-env tsc --project tsconfig.build.json --outDir dist/amd --module amd
[0] npm ERR! You can get information on how to open an issue for this project with:
[0] npm ERR!     npm bugs aurelia-ux
[0] npm ERR! Or if that isn't available, you can get their info via:
[0] npm ERR!     npm owner ls aurelia-ux
[0] npm ERR! There is likely additional logging output above.
```

**Expected/desired behavior:**
Expected all tests to pass and build to succeed.

I'm still getting errors when trying to build, I checked out a fresh copy of Master a few moments ago, same errors as previously reported [https://github.com/aurelia/ux/issues/30#issue-198820069](https://github.com/aurelia/ux/issues/30#issue-198820069)

```
[3] src/effects/paper-wave.ts(342,5): error TS2531: Object is possibly 'null'.
[3] src/styles/style-controller.ts(60,5): error TS2322: Type 'Node | null' is not assignable to type 'Node'.
[3]   Type 'null' is not assignable to type 'Node'.
[3] src/styles/style-resource.ts(34,5): error TS2322: Type 'Promise<StyleFactory>' is not assignable to type 'Promise<StyleResource>'.
[3]   Type 'StyleFactory' is not assignable to type 'StyleResource'.
[3]     Property 'styleObjectType' is private in type 'StyleFactory' but not in type 'StyleResource'.
```
Following is from npm-debug.log 

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
2 info using npm@4.1.1
3 info using node@v7.2.1
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle aurelia-ux@0.1.15~prebuild: aurelia-ux@0.1.15
6 verbose lifecycle aurelia-ux@0.1.15~prebuild: unsafe-perm in lifecycle true
7 verbose lifecycle aurelia-ux@0.1.15~prebuild: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/Users/account/git/ux-master/node_modules/.bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet
8 verbose lifecycle aurelia-ux@0.1.15~prebuild: CWD: /Users/account/git/ux-master
9 silly lifecycle aurelia-ux@0.1.15~prebuild: Args: [ '-c', 'cross-env rimraf dist' ]
10 silly lifecycle aurelia-ux@0.1.15~prebuild: Returned: code: 0  signal: null
11 info lifecycle aurelia-ux@0.1.15~build: aurelia-ux@0.1.15
12 verbose lifecycle aurelia-ux@0.1.15~build: unsafe-perm in lifecycle true
13 verbose lifecycle aurelia-ux@0.1.15~build: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/Users/account/git/ux-master/node_modules/.bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet
14 verbose lifecycle aurelia-ux@0.1.15~build: CWD: /Users/account/git/ux-master
15 silly lifecycle aurelia-ux@0.1.15~build: Args: [ '-c',
15 silly lifecycle   'concurrently "npm run build:amd" "npm run build:commonjs" "npm run build:es2015" "npm run build:system"' ]
16 silly lifecycle aurelia-ux@0.1.15~build: Returned: code: 1  signal: null
17 info lifecycle aurelia-ux@0.1.15~build: Failed to exec build script
18 verbose stack Error: aurelia-ux@0.1.15 build: `concurrently "npm run build:amd" "npm run build:commonjs" "npm run build:es2015" "npm run build:system"`
18 verbose stack Exit status 1
18 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
18 verbose stack     at emitTwo (events.js:106:13)
18 verbose stack     at EventEmitter.emit (events.js:191:7)
18 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:40:14)
18 verbose stack     at emitTwo (events.js:106:13)
18 verbose stack     at ChildProcess.emit (events.js:191:7)
18 verbose stack     at maybeClose (internal/child_process.js:885:16)
18 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
19 verbose pkgid aurelia-ux@0.1.15
20 verbose cwd /Users/account/git/ux-master
21 error Darwin 15.5.0
22 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
23 error node v7.2.1
24 error npm  v4.1.1
25 error code ELIFECYCLE
26 error aurelia-ux@0.1.15 build: `concurrently "npm run build:amd" "npm run build:commonjs" "npm run build:es2015" "npm run build:system"`
26 error Exit status 1
27 error Failed at the aurelia-ux@0.1.15 build script 'concurrently "npm run build:amd" "npm run build:commonjs" "npm run build:es2015" "npm run build:system"'.
27 error Make sure you have the latest version of node.js and npm installed.
27 error If you do, this is most likely a problem with the aurelia-ux package,
27 error not with npm itself.
27 error Tell the author that this fails on your system:
27 error     concurrently "npm run build:amd" "npm run build:commonjs" "npm run build:es2015" "npm run build:system"
27 error You can get information on how to open an issue for this project with:
27 error     npm bugs aurelia-ux
27 error Or if that isn't available, you can get their info via:
27 error     npm owner ls aurelia-ux
27 error There is likely additional logging output above.
28 verbose exit [ 1, true ]
```

--- a/package.json
+++ b/package.json
@@ -21,35 +21,35 @@
     "url": "https://github.com/aurelia/ux"
   },
   "scripts": {
+    "postinstall": "npm uninstall @types/fs-extra @types/handlebars @types/highlight.js @types/lodash @types/marked @types/minimatch @types/shelljs",
     "lint": "cross-env tslint --project tsconfig.json",
-
     "pretest": "cross-env npm run lint",
-    "test": "cross-env tsc && karma start --single-run",
-    "test-watch": "concurrently \"./node_modules/.bin/tsc --watch\" \"./node_modules/.bin/karma start\"",
-
+    "test": "cross-env rimraf dist && tsc && karma start --single-run",
+    "develop": "concurrently \"./node_modules/.bin/tsc --watch\" \"./node_modules/.bin/karma start\"",
     "prebuild:amd": "cross-env rimraf dist/amd",
     "build:amd": "cross-env tsc --project tsconfig.build.json --outDir dist/amd --module amd",
     "postbuild:amd": "cross-env copyfiles --up 1 src/**/*.html src/**/*.css dist/amd",
-
     "prebuild:commonjs": "cross-env rimraf dist/commonjs",
     "build:commonjs": "cross-env tsc --project tsconfig.build.json --outDir dist/commonjs --module commonjs",
     "postbuild:commonjs": "cross-env copyfiles --up 1 src/**/*.html src/**/*.css dist/commonjs",
-
     "prebuild:es2015": "cross-env rimraf dist/es2015",
-    "build:es2015": "cross-env tsc --project tsconfig.build.json --outDir dist/es2015 --module es2015",
+    "build:es2015": "cross-env tsc --project tsconfig.build.json --outDir dist/es2015 --module es2015 --target es2015",
     "postbuild:es2015": "cross-env copyfiles --up 1 src/**/*.html src/**/*.css dist/es2015",
-
+    "prebuild:native-modules": "cross-env rimraf dist/native-modules",
+    "build:native-modules": "cross-env tsc --project tsconfig.build.json --outDir dist/native-modules --module es2015",
+    "postbuild:native-modules": "cross-env copyfiles --up 1 src/**/*.html src/**/*.css dist/native-modules",
     "prebuild:system": "cross-env rimraf dist/system",
     "build:system": "cross-env tsc --project tsconfig.build.json --outDir dist/system --module system",
     "postbuild:system": "cross-env copyfiles --up 1 src/**/*.html src/**/*.css dist/system",
-
     "prebuild": "cross-env rimraf dist",
-    "build": "concurrently \"npm run build:amd\" \"npm run build:commonjs\" \"npm run build:es2015\" \"npm run build:system\"",
-    "postbuild": "npm run build:doc",
-
-    "prebuild:doc": "cross-env rimraf doc/api.json && rimraf dist/doc-temp && tsc --project tsconfig.build.json --outFile dist/doc-temp/index.js && node doc/shape-defs",
-    "build:doc": "cross-env typedoc --json doc/api.json --excludeExternals --includeDeclarations --mode modules --target ES6 --name aurelia-ux-docs --ignoreCompilerErrors --tsconfig doc/tsconfig.json dist/doc-temp/",
-    "postbuild:doc": "cross-env node doc/shape-doc && rimraf dist/doc-temp"
+    "build": "concurrently \"npm run build:amd\" \"npm run build:commonjs\" \"npm run build:es2015\" \"npm run build:native-modules\" \"npm run build:system\"",
+    "predoc": "cross-env rimraf doc/api.json && rimraf dist/doc-temp && tsc --project tsconfig.build.json --outFile dist/doc-temp/aurelia-ux.js && node doc/shape-defs && copyfiles tsconfig.json dist/doc-temp",
+    "doc": "cross-env typedoc --json doc/api.json --excludeExternals --includeDeclarations --mode modules --target ES6 --name aurelia-ux-docs dist/doc-temp/",
+    "postdoc": "cross-env node doc/shape-doc && rimraf dist/doc-temp",
+    "changelog": "cross-env conventional-changelog -p ux -i doc/CHANGELOG.md -s",
+    "bump-version": "npm --no-git-tag-version version",
+    "preprepare-release": "cross-env npm run test",
+    "prepare-release": "cross-env npm run changelog && npm run build && npm run doc"
   },
   "jspm": {
     "registry": "npm",
@@ -91,25 +91,28 @@
     "aurelia-framework": "^1.0.0"
   },
   "devDependencies": {
-    "aurelia-bootstrapper": "^1.0.0",
-    "aurelia-pal-browser": "^1.0.0",
-    "aurelia-polyfills": "^1.1.0",
-    "aurelia-testing": "^1.0.0-beta.2.0.0",
-    "concurrently": "^2.2.0",
-    "copyfiles": "^1.0.0",
-    "cross-env": "^2.0.1",
-    "jasmine-core": "^2.4.1",
-    "karma": "^1.2.0",
+    "@types/node": "^7.0.11",
+    "aurelia-bootstrapper": "^2.1.1",
+    "aurelia-pal-browser": "^1.2.0",
+    "aurelia-pal-nodejs": "^1.0.0-beta.1.0.0",
+    "aurelia-polyfills": "^1.2.1",
+    "aurelia-testing": "^1.0.0-beta.3.0.1",
+    "concurrently": "^3.4.0",
+    "conventional-changelog-cli": "^1.3.1",
+    "copyfiles": "^1.2.0",
+    "cross-env": "^3.2.4",
+    "jasmine-core": "^2.5.2",
+    "karma": "^1.5.0",
     "karma-chrome-launcher": "^2.0.0",
     "karma-ie-launcher": "^1.0.0",
-    "karma-jasmine": "^1.0.2",
-    "karma-requirejs": "^1.0.0",
-    "requirejs": "^2.2.0",
-    "requirejs-text": "^2.0.12",
-    "rimraf": "^2.5.4",
-    "tslint": "^3.15.1",
-    "typedoc": "^0.5.0",
-    "typescript": "^2.0.0"
+    "karma-jasmine": "^1.1.0",
+    "karma-requirejs": "^1.1.0",
+    "requirejs": "^2.3.3",
+    "requirejs-text": "^2.0.15",
+    "rimraf": "^2.6.1",
+    "tslint": "^4.4.2",
+    "typedoc": "^0.5.9",
+    "typescript": "^2.2.1"
   },
   "aurelia": {
     "build": {

--- a/src/effects/paper-wave.ts
+++ b/src/effects/paper-wave.ts
@@ -339,7 +339,7 @@ export class PaperWave {
    * @returns Current instance for method chaining.
    */
   public remove() {
-    this.$.parentNode.removeChild(this.$);
+    (this.$.parentNode as Node).removeChild(this.$);
     return this;
   }
 }

--- a/src/styles/style-controller.ts
+++ b/src/styles/style-controller.ts
@@ -57,7 +57,7 @@ export class StyleController {
   }
 
   private removeStyleElement() {
-    this.styleElementParent = this.styleElement.parentNode;
+    this.styleElementParent = this.styleElement.parentNode as Node;
     DOM.removeNode(this.styleElement);
     this.onRemove();
   }

--- a/src/styles/style-resource.ts
+++ b/src/styles/style-resource.ts
@@ -34,7 +34,7 @@ export class StyleResource {
     return styleStrategy.loadStyleFactory(container, this.styleObjectType).then(styleFactory => {
       this.factory = styleFactory;
       this.hooks.factory = this.factory;
-      return styleFactory;
+      return this;
     });
   }
 }

